Resolve high severity Sonar issues

nikomakela opened the pull request:
TH-1441.

nikomakela committed:
style(sonar): use spread instead object.assign

TH-1441

nikomakela committed:
style(sonar): eventsFederationApolloClient and utils

TH-1441.

Sonar issues:
- 'object.internalId' may use Object's default stringification format
('[object Object]') when stringified.
- Use `for…of` instead of `.forEach(…)`.
- Refactor this function to not always return the same value.
- Prefer `globalThis.window` over `window`.
- Prefer using nullish coalescing operator (`??=`) instead of an
assignment expression, as it is simpler to read.

nikomakela committed:
style(sonar): eventInfo

TH-1441.

Sonar issue: Provide a compare function that depends on
"String.localeCompare", to reliably sort elements alphabetically.

nikomakela committed:
style(sonar): usePromise tests

TH-1441.

Sonar issue (high severity): "Refactor this code to not nest functions
more than 4 levels deep."

nikomakela committed:
style(sonar): eventUtils

TH-1441.

Sonar issues:
- Provide a compare function that depends on "String.localeCompare", to
reliably sort elements alphabetically. High severity.
- Default parameters should be last.
- Use the "RegExp.exec()" method instead.
- Prefer `String#replaceAll()` over `String#replace()`.

nikomakela committed:
style(sonar): getDateRangeStr

TH-1441.

Sonar issue (high severity): "Refactor this function to reduce its
Cognitive Complexity from 16 to the 15 allowed."

nikomakela committed:
style(sonar): headless cms service

TH-1441.

Sonar issues (high severity):
- Refactor this code to not nest functions more than 4 levels deep.
- Provide a compare function that depends on "String.localeCompare", to
reliably sort elements alphabetically.

nikomakela committed:
style(sonar): cms helper provider

TH-1441.

Sonar issue: The 'context' object passed as the value prop to the
Context provider changes every render. To fix this consider wrapping it
in a useMemo hook.

nikomakela committed:
style(sonar): askem

TH-1441.

Sonar issue: Prefer using an optional chain expression instead, as it's
more concise and easier to read.

nikomakela committed:
style(sonar): date range picker

TH-1441.

Sonar issue: Extract this nested ternary operation into an independent
statement.

nikomakela committed:
style(sonar): dateselector stories

TH-1441.

Sonar issue: The return value of "toLocaleDateString" must be used.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr877.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
style(sonar): fix type check issues

TH-1441

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr877.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr877.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr877.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr877.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr877.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders committed:
fix(types): make use of AppLanguage more in a typesafe way

done:
- remove all "as AppLanguage" typecasts from codebase
- use AppLanguage instead of string as a type where applicable
- if locale/language is string, make sure it actually is an AppLanguage
  by using isAppLanguage typeguard first, and only after that use it as
  an AppLanguage
- use DEFAULT_LANGUAGE (i.e. "fi") as a fallback for AppLanguage values
- make slightly better use of typescript's type system in isAppLanguage
  typeguards by instead of downcasting unknown to AppLanguage upcast
  `('fi' | 'en' | 'sv')[]` to `readonly string[]` and narrow `unknown`
  to `string` first

refs TH-1441

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr877.dev.hel.ninja :laughing::tada::tada::tada:
